src/final_chain/final_chain.cpp

```diff
@@ -22,7 +22,6 @@ class FinalChainImpl final : public FinalChain {
   mutable shared_ptr<BlockHeader> last_block_;
 
   util::ThreadPool executor_thread_{1};
```

Comment:
I would add comment here, that this TP can not have more than 1 thread

Reply:
Just added the comment after discussions with @JakubFornadel